dotnet8/hello-native-aot/{{cookiecutter.project_name}}/template.yaml

```diff
@@ -0,0 +1,48 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: >
+  Sample SAM Template for {{ cookiecutter.project_name }}
+
+# More info about Globals: https://github.com/awslabs/serverless-application-model/blob/master/docs/globals.rst
+Globals:
+  Function:
+    Timeout: 10
+    MemorySize: 512
+
+Resources:
+  HelloWorldAotFunction:
+    Type: AWS::Serverless::Function # More info about Function Resource: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction
+    Properties:
+      CodeUri: ./src/HelloWorldAot/
+      Handler: bootstrap
+      Runtime: dotnet8
```

Comment:
Shouldn't this (AOT) be using a provided runtime with dotnet as build method?

Reply:
We can actually set the Runtime to dotnet8 instead of provided this time (couldn't for .NET 7 since that wasn't a managed runtime). 

Since we're setting the AssemblyName and Handler to bootstrap, the lambda startup code will find that and just execute it and the .NET 8 runtime that is included in the managed runtime won't be used. 

It's worth using the .NET 8 managed runtime because it will have the necessary system libs already installed that provided won't have (e.g libicu).

Reply:
Thanks for explaining in detail. Should we put a warning message in SAM CLI if a customer tries to use provided runtime with dotnet8 build method? Or are we already blocking that?

Reply:
So if a user used runtime `provided.al2023` or `provided.al2` with a .NET 8 project, they'd need to use build method `dotnet7` (which would be weird) since there is no custom property build method for dotnet8 (only runtime dotnet8, but your runtime would be provided.al2023). The build method `dotnet7` actually just calls a normal `dotnet lambda package` and isn't specific to .NET 7 as long as the customer has .NET 8 on their machine. 

On one hand, they would hopefully not use `dotnet7` build method for .NET 8, but on the other hand, it's a simple change to change the project file to point to .NET 8 and rebuild, so it could happen. I tried doing just that but get a serialization error due to reflection (below).

I can also mention the correct upgrade path in the blog as a mitigation. 

```
System.InvalidOperationException: Reflection-based serialization has been disabled for this application. Either use the source generator APIs or explicitly configure the 'JsonSerializerOptions.TypeInfoResolver' property.
at System.Text.Json.ThrowHelper.ThrowInvalidOperationException_JsonSerializerIsReflectionDisabled() + 0x30
at System.Text.Json.JsonSerializerOptions.ConfigureForJsonSerializer() + 0x44
at System.Text.Json.JsonSerializer.GetTypeInfo(JsonSerializerOptions, Type) + 0x34
at System.Text.Json.JsonSerializer.GetTypeInfo[T](JsonSerializerOptions) + 0x2c
at HelloWorld.Function.<FunctionHandler>d__3.MoveNext() + 0x148
```